src/Symfony/Component/Security/Core/Tests/Authorization/ExpressionLanguageTest.php

```diff
@@ -51,6 +52,52 @@ public function provider()
         $roles = ['ROLE_USER', 'ROLE_ADMIN'];
         $user = new User('username', 'password', $roles);
 
+        $anonymousToken = new AnonymousToken('firewall', 'anon.');
+        $rememberMeToken = new RememberMeToken($user, 'providerkey', 'firewall');
+        $usernamePasswordToken = new UsernamePasswordToken('username', 'password', 'providerkey', $roles);
+
+        return [
+            [$anonymousToken, "is_granted('IS_AUTHENTICATED_ANONYMOUSLY')", true],
+            [$anonymousToken, "is_granted('IS_AUTHENTICATED_FULLY')", false],
+            [$anonymousToken, "is_granted('IS_ANONYMOUS')", true],
+            [$anonymousToken, "is_granted('IS_REMEMBERED')", false],
+            [$anonymousToken, "is_granted('ROLE_USER')", false],
+
+            [$rememberMeToken, "is_granted('IS_AUTHENTICATED_ANONYMOUSLY')", true],
+            [$rememberMeToken, "is_granted('IS_AUTHENTICATED_FULLY')", false],
+            [$rememberMeToken, "is_granted('IS_ANONYMOUS')", false],
+            [$rememberMeToken, "is_granted('IS_REMEMBERED')", true],
+            [$rememberMeToken, "is_granted('ROLE_FOO')", false],
+            [$rememberMeToken, "is_granted('ROLE_USER')", true],
+
+            [$usernamePasswordToken, "is_granted('IS_AUTHENTICATED_ANONYMOUSLY')", true],
+            [$usernamePasswordToken, "is_granted('IS_AUTHENTICATED_FULLY')", true],
+            [$usernamePasswordToken, "is_granted('IS_ANONYMOUS')", false],
+            [$usernamePasswordToken, "is_granted('IS_REMEMBERED')", false],
+            [$usernamePasswordToken, "is_granted('ROLE_FOO')", false],
+            [$usernamePasswordToken, "is_granted('ROLE_USER')", true],
+        ];
+    }
+
+    /**
+     * @dataProvider provideLegacyIsAuthenticated
+     */
```

Comment:
shouldn't it be marked a `@group legacy` with `@expectedDeprecation`?

Reply:
Thanks, fixed (And also fixed this test, as it wasn't using the data provider at all :see_no_evil: )